Let textareas resize automatically without having to highlight placeholders

quis opened the pull request:
Scrolling within `<textarea>`s on the page is a bit grim. Which is why we don’t do it for the textboxes that people use to edit templates.

This pull request extends the auto-resizing of `<textarea>`s to those which don’t need the highlighting of placeholders.

***

![awanp](https://user-images.githubusercontent.com/355079/67001939-b7dca580-f0d2-11e9-94fc-eb6b07a0df44.gif)

***

![image](https://user-images.githubusercontent.com/355079/67003352-a0eb8280-f0d5-11e9-9f1a-81e57129ccd7.png)

tombye commented on app/assets/javascripts/enhancedTextbox.js:
```diff
@@ -0,0 +1,85 @@
+
+    };
+
+    this.contentEscaped = () => $('<div/>').text(this.$textbox.val()).html();
```
Is the content escaped here? I can't find reference to it in the docs for `.val()`.

quis replied:
There some detail in this commit: https://github.com/alphagov/notifications-admin/commit/836fe77c372eaa10bf6273b0efef54f6bcd709bb

TL;DR is that it’s the `$('<div/>').text(…)` part that does the escaping.

tombye replied:
Gotcha, sorry, should have checked those docs.

quis commented on app/assets/javascripts/enhancedTextbox.js:
```diff
@@ -0,0 +1,85 @@
+      let visibleTextbox;
+
+      this.highlightPlaceholders = (
+        typeof textarea.data('highlightPlaceholders') === 'undefined' ||
```
`this.highlightPlaceholders` should be `true` if:
- `data-highlightPlaceholders` is explicitly set to `"true"`
- `data-highlightPlaceholders` is not set (i.e. it’s true by default)

tombye replied:
Right, I didn't know the second point but see there's a test for it so 👍

quis commented:
@tombye I fixed the calls to the textbox macro, and added a test to make sure the right data attributes are present on the edit templates page: https://github.com/alphagov/notifications-admin/compare/resize-all-textbox-dem#diff-f2eb304b93cc383727c0ab7fc8fbd464R349-R350
